### PR TITLE
Initial version of customer service.

### PR DESCRIPTION
**Pull request by dartartem**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+		fmt.Sprintf("{\"ID\" : \"%v\", \"PARTITION_ID\" : %v, \"DESTINATION\" : %v, \"event-type\" : %v, \"event-aggregate-type\" : %v, \"event-aggregate-id\" : %v}",
+			id, aggregateId, aggregateType, eventType, aggregateType, aggregateId),
+		payload,
+		0)	
+}
+
+func reserveCredit(customerId int64, orderId int64, orderTotal int64) {
+	customer := Customer{}
+
+	err := dataBase.First(&customer, customerId).Error
```
Comment (cer) [inline main.go]: This needs to be done in a DB transaction

File: main.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+func publishEvent(domainEvent interface{}, aggregateType string, aggregateId int64, eventType string) {
+	id := generateId()	
+
+    payload, err := json.Marshal(domainEvent)
+    
+	if err != nil {
+        log.Fatal(err)
+    }
+
+	dataBase.Exec("insert into eventuate.message (id, destination, headers, payload, published, creation_time) values (?, ?, ?, ?, ?, ROUND(UNIX_TIMESTAMP(CURTIME(4)) * 1000))",
```
Comment (cer) [inline main.go]: This should use DB generated IDs

**Review by cer — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (dartartem) [reply main.go]: yes, I did not finish, this is in todo list, as I wrote in report.

File: main.go (hunk 2 above)
Comment (dartartem) [reply main.go]: I am planned to do so, as I wrote in my report

**Pull request merged**